dev-tools/repo-depot-standalone/src/main.rs

```diff
@@ -121,33 +159,50 @@ impl RepoMetadata {
 
     pub fn load_repo(
         &mut self,
-        omicron_repo: OmicronRepo,
+        plan: ArtifactsWithPlan,
+        log: &Logger,
     ) -> anyhow::Result<()> {
         let repo_index = self.repos.len();
 
-        let tuf_repo = omicron_repo.repo();
-        for (target_name, target) in &tuf_repo.targets().signed.targets {
-            let target_hash: &[u8] = &target.hashes.sha256;
-            let target_hash_array: [u8; 32] = target_hash
-                .try_into()
-                .context("sha256 hash wasn't 32 bytes")?;
-            let artifact_hash = ArtifactHash(target_hash_array);
-            self.targets_by_hash
-                .insert(artifact_hash, (repo_index, target_name.clone()));
+        for artifact_meta in &plan.description().artifacts {
+            let artifact_hash = artifact_meta.hash;
+            let artifact_id = &artifact_meta.id;
+            let artifact_hash_id = ArtifactHashId {
+                kind: artifact_id.kind.clone(),
+                hash: artifact_hash,
+            };
+            if let Some((_, old_hash_id)) = self
+                .targets_by_hash
+                .insert(artifact_meta.hash, (repo_index, artifact_hash_id))
+            {
+                warn!(
+                    log,
+                    "artifact hash found multiple times";
+                    "hash" => %artifact_hash,
+                    "new_kind" => %artifact_id.kind,
+                    "name" => %artifact_id.name,
+                    "old_kind" => %old_hash_id.kind,
+                );
```

Comment:
This is expected with fake TUF repos. It might alarm someone new using this. Perhaps we can add a note or something that it's OK if you're using a fake repo?